core/store/src/trie/trie_storage.rs

```diff
@@ -90,14 +90,21 @@ struct TrieCacheMetrics {
 }
 
 impl TrieCacheInner {
+    /// Assumed number of bytes used to store an entry in the cache.
+    ///
+    /// 100 bytes is an approximation based on lru 0.7.5.
+    pub(crate) const PER_ENTRY_OVERHEAD: u64 = 100;
+
     pub(crate) fn new(
-        cache_capacity: usize,
         deletions_queue_capacity: usize,
         total_size_limit: u64,
         shard_id: ShardId,
         is_view: bool,
     ) -> Self {
-        assert!(cache_capacity > 0 && total_size_limit > 0);
+        assert!(total_size_limit > 0);
+        // upper bound on capacity for allocation purposes only
+        let cache_capacity =
+            (total_size_limit + Self::PER_ENTRY_OVERHEAD - 1) / Self::PER_ENTRY_OVERHEAD;
```

Comment:
Interesting! So in practice I think this means we always end up over-allocating the cache? 

There's perhaps something smarter here (like dividing by 2 * per entry overhead) I don't immediately see an obviously better appraoch.

Reply:
seems to make much more sense to just use unbounded lru cache now